src/main/java/picard/illumina/parser/readers/CbclReader.java

```diff
@@ -269,7 +272,8 @@ private void advance() {
             for (int cycle = 0; cycle < outputLengths[read]; cycle++) {
                 CycleData currentCycleData = cycleData[totalCycleCount];
 
-                if (cachedTilePosition[totalCycleCount] >= cachedTile[totalCycleCount].length) {
+                if (cachedTilePosition[totalCycleCount] >= cachedTile[totalCycleCount].length
+                        || cachedTilePosition[totalCycleCount] >= cycleData[totalCycleCount].getTileInfo().getNumClustersInTile()) {
```

Comment:
This is for the case where you have a odd number of PF reads despite the fact that you have to have a even number since there are 2 reads per byte. Perhaps I could do away with the first check in the or.. this may be a bit belt and suspenders.